src/backends/amqp.ts

```diff
@@ -73,18 +73,28 @@ export default class AMQPBackend implements CeleryBackend {
           .then(() => Promise.resolve(ch))
       )
       .then(ch =>
+        
         ch.publish(
           "",
           queue,
           Buffer.from(
-            JSON.stringify({
-              status: state,
-              result,
-              traceback: null,
-              children: [],
-              task_id: taskId,
-              date_done: new Date().toISOString()
-            })
+            state == 'FAILURE' ?
+              JSON.stringify({
+                status: state,
+                result,
+                traceback: null,
+                children: [],
+                task_id: taskId,
+                date_done: new Date().toISOString()
+              }) :
+              JSON.stringify({
+                status: state,
+                result,
+                traceback: null,
+                children: [],
+                task_id: taskId,
+                date_done: new Date().toISOString()
+              })
```

Comment:
As I see `result` should be null when state is failure
```suggestion
              JSON.stringify({
                status: state,
                state == 'FAILURE' ? null : result,
                traceback: null,
                children: [],
                task_id: taskId,
                date_done: new Date().toISOString()
              })
```